core/modules/replicate.h

```diff
@@ -0,0 +1,36 @@
+#ifndef BESS_MODULES_REPLICATE_H_
+#define BESS_MODULES_REPLICATE_H_
+
+#include "../module.h"
+#include "../module_msg.pb.h"
+
+#define MAX_REPL_GATES 20
```

Comment:
Instead of using a macro, how about `static const gate_idx_t kMaxGates` as a public member of `class Replicate`?